From 637c177942eabbbbd5d28a83b281dc7079661093 Mon Sep 17 00:00:00 2001
From: Dan Prince <dprince@redhat.com>
Date: Mon, 12 Jan 2015 15:56:44 -0500
Subject: [PATCH] add /usr/share/$project/*-dist.conf to the default

From 50ae3e476a3c670e6796a5caa9dd2fd69c7ef931 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <pbrady@redhat.com>
Date: Mon, 11 Feb 2013 17:32:55 +0000
Subject: [PATCH] add /usr/share/$project/*-dist.conf to the default
config set

This will lookup /usr/share/$project/$project-dist.conf
and /usr/share/$project/$prog-dist.conf
in the correct precedence order by default.
For example from command line utils which don't specify
the --config-file option.  Note if daemon init scripts
explicitly specify a --config-file then they must specify
all required.

For reference the precedence order is that later --config-file
options will override settings from previous ones.
So when no --config-file option is specified, we insert
the "dist" config file as the first to be parsed,
and settings from there are overridden by any from
/etc/$project/$project.conf and /etc/$prog.conf respectively.

For completeness, note settings in configs from --config-dir
take precedence over any of the --config-file settings above.
---
 oslo_config/cfg.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/oslo_config/cfg.py b/oslo_config/cfg.py
index c44de89..e3e7b65 100644
--- a/oslo_config/cfg.py
+++ b/oslo_config/cfg.py
@@ -549,6 +549,10 @@ def find_config_files(project=None, prog=None, extension='.conf'):
 
     config_files = []
     if project:
+        config_files.append(_search_dirs(['/usr/share/%s/' % project],
+                                         project, '-dist%s' % extension))
+        config_files.append(_search_dirs(['/usr/share/%s/' % project],
+                                         prog, '-dist%s' % extension))
         config_files.append(_search_dirs(cfg_dirs, project, extension))
     config_files.append(_search_dirs(cfg_dirs, prog, extension))
 
-- 
1.9.3

